src/python_testing/TC_AVSM_2_1.py

```diff
@@ -161,276 +164,443 @@ async def test_TC_AVSM_2_1(self):
                 endpoint=endpoint, cluster=cluster, attribute=attr.MaxConcurrentEncoders
             )
             logger.info(f"Rx'd MaxConcurrentEncoders: {value}")
+            asserts.assert_is_not_none(value, "MaxConcurrentEncoders is None")
+            matter_asserts.assert_valid_uint8(value, "MaxConcurrentEncoders")
 
         self.step(3)
         if self.pics_guard(self.check_pics("AVSM.S.A0001")):
```

Comment:
Please replace **ALL** of these PICS guards with:

```
if await self.attribute_guard(endpoint=endpoint, attribute=attr.ATTR_NAME_HERE):
```

So:

```
        if self.pics_guard(self.check_pics("AVSM.S.A0001")):
            value = await self.read_single_attribute_check_success(
                endpoint=endpoint, cluster=cluster, attribute=attr.MaxEncodedPixelRate
            )
```

becomes:

```
        if await self.attribute_guard(endpoint=endpoint, attribute=attr.MaxEncodedPixelRate):
            value = await self.read_single_attribute_check_success(
                endpoint=endpoint, cluster=cluster, attribute=attr.MaxEncodedPixelRate
            )
```

This will remove dependies on PICS for this test, and make it more likely to catch issues.

Reply:
Thank you for the review. we have fixed it.